### PR TITLE
Create support bundle from failed and partiallyfailed snapshots

### PR DESCRIPTION
**Pull request by marccampbell**
This PR adds an informer to velero's backup types. When the status changes to Failed or PartiallyFailed, if this is a kots generate backup, kots will now generate a support bundle using a specific collectors and analyzers and link to troubleshooting from the failed snapshot.

There are still some things to do:

1. We need a set of collectors.  
1. We need analyzers
1. The informer starts when kotsadm starts. This works, but will likely be a problem if velero is not installed and then is installed later. We need the "detect velero" to restart the informer, if the informer didn't start at startup

### CODE REVIEW

File: pkg/apiserver/server.go [hunk 1]
```diff
@@ -21,6 +22,10 @@ func Start() {
 		log.Println("Failed to run migrations", err)
 	}
 
+	if err := informers.Start(); err != nil {
+		log.Println("Failed to start informers", err)
```
Comment (divolgin) [inline pkg/apiserver/server.go:26]: no panic here?

File: pkg/informers/informers.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+						appID, ok := backup.Annotations["kots.io/app-id"]
+						if !ok {
+							logger.Errorf("failed to find app id anotation on backup")
+						}
+
+						backup.Annotations["kots.io/support-bundle-requested"] = time.Now().UTC().Format(time.RFC3339)
+
+						if _, err := veleroClient.Backups(backup.Namespace).Update(backup); err != nil {
+							logger.Error(err)
+							break
```
Comment (divolgin) [inline pkg/informers/informers.go:64]: i'd think this should be a continue

File: pkg/apiserver/server.go (hunk 1 above)
Comment (marccampbell) [reply pkg/apiserver/server.go:26]: what about if there's no velero?

Comment (divolgin) [reply pkg/apiserver/server.go:26]: yeah... i guess it's good enough for now

**Pull request merged**